### PR TITLE
Added as.caretList function

### PR DESCRIPTION
**Pull request by washcycle**
Added ability to convert list of caret models into a caretList object.

### CODE REVIEW

**Comment by zachmayer** [comment]
I don't like is.caretList changing the object's type.

I'd prefer a new as.caretList function.

**Comment by washcycle** [comment]
Hmmm this is a` as.caretList` function? Are we on the same page here? The reason I wanted this was because I trained several models and one of them didn't work out. I wanted to drop it.

I could subset the models by `models[1:3]` dropping the reaming model(s). However to ensemble them they needed to be of type caretList. I looked at how the caretList was created which led me to write this function which worked great.

**Comment by zachmayer** [comment]
I think we could write a `[` method for `caretList` that doesn't drop the class on subsetting.

**Comment by washcycle** [comment]
I like that idea

File: R/caretList.R [hunk 1]
```diff
@@ -255,6 +255,18 @@ is.caretList <- function(object){
...
+#' @description Convert list of Caret Models to caretList object
+#' @param modelList
+#' @export
+as.caretList <- function(modelList){
+  if(class(modelList) != "list"){
+    stop("modelList must be a list of caret models")
+  }
+  # TODO consider other parammeter checking perhaps check for the train class in the list??
+  class(modelList) <- "caretList"
+}
```
Comment (zachmayer) [inline R/caretList.R]: please add an explicit `return(modelList)`

File: R/caretList.R [hunk 2]
```diff
@@ -255,6 +255,18 @@ is.caretList <- function(object){
...
 
+#' @title Convert list of Caret Models to caretList object
+#' @description Convert list of Caret Models to caretList object
+#' @param modelList
+#' @export
+as.caretList <- function(modelList){
+  if(class(modelList) != "list"){
+    stop("modelList must be a list of caret models")
+  }
+  # TODO consider other parammeter checking perhaps check for the train class in the list??
```
Comment (zachmayer) [inline R/caretList.R]: Let's add a check that all the objects in the list are of class train.  Something like `all(sapply(modelList, is.train))`

**Comment by zachmayer** [comment]
2 more notes:
1. It looks like `as` is an `S4` method, so we should write an S4 method here.  I dunno how to do that, but I'll do some research.
2. Please add a test

**Comment by washcycle** [comment]
As is an S4 method, but as.classname is an S3 method. `getS3method('as.data.frame', 'data.frame')` I'll probably go and stop being lazy and clone and make these changes and add a test(s) this weekend. I'll make a new request once i'm done and close/references these ones out.

**Comment by zachmayer** [comment]
That would be perfect, thanks!

**Pull request closed without merging**